R/PipeOpPredClassifSurv.R

```diff
@@ -0,0 +1,79 @@
+
+PipeOpPredClassifSurv <- R6Class(
+  "PipeOpPredClassifSurv",
+  inherit = PipeOpPredTransformer,
+
+  public = list(
+    initialize = function(id = "trafopred_classifsurv") {
+      super$initialize(id = id,
+                       input = data.table::data.table(name = c("input", "meta"), train = c("NULL", "*"), predict = c("PredictionClassif", "*")),
+                       output = data.table::data.table(name = "output", train = "NULL", predict = "PredictionSurv"))
+    }
+  ),
+
+  private = list(
+    .train = function(input) {
+      self$state = list()
+      list(input)
+    },
+
+    .predict = function(input) {
+      data = input[[2]]
+      pred = input[[1]]
+      data <- cbind(data, pred = pred$prob[, 2])
+
+      ## convert hazards to surv as prod(1 - h(t))
+      surv <- t(sapply(unique(data$id), function(i) {
+        x <- cumprod((1 - data[data$id == i, "pred"]))
+        x
+      }))
+
+      r <- list()
+      time <- sort(unique(data$tend))
+      ## coerce to distribution and crank
+      r <- .surv_return(time, surv = surv)
+
+      crank = c()
+      distr = list()
+
+      # select real tend values by only selecting the last row of each id
+      # basically a slightly more complex unique()
+      real_tend <- data$time2[seq_len(nrow(data)) %% sum(data$id == 1) == 0]
+
+      ## create prediction object
+      data = data %>% group_by(id) %>% filter(tend == max(tend))
```

Comment:
1. For every library that is not in DESCRIPTION and we use a function from it, do `library`::`function(...)` eg from `stats::somefun()`. 
2. Always use `|>` instead of `%>%` since the first is included in base `R` and does the same thing. 
3. Avoid using `group_id` and `filter` functions (from `dplyr` I guess). Here you are trying to get the `data[["ped_status"]]` for every subject and the last `tend`, take only what you need if possibel in base R, like you did in the previous expression